src/components/Stations/components/StationTableColumns.tsx

```diff
@@ -0,0 +1,165 @@
+import BaseWeatherIcon from "@/components/BaseComponents/BaseWeatherIcon";
+import CommonButton from "@/components/Common/CommonButton";
+import SvgInline from "@/components/Common/SvgInline";
+import { Measurements, WeatherConditions, WeatherDataResponse } from "@/types";
+import { HeartIcon } from "@heroicons/react/24/solid";
+import { createColumnHelper, SortingFn } from "@tanstack/react-table";
+import { fullDateWithTime } from "../../../utils/dateTimeUtils";
+
+// Sorting Fns
+const sortStatusFn: SortingFn<WeatherDataResponse> = (rowA, rowB) => {
+    return rowA.original.weather_station_id.name.localeCompare(
+        rowB.original.weather_station_id.name
+    );
+};
+const sortTemperatureFn: SortingFn<WeatherDataResponse> = (rowA, rowB) => {
+    return rowA.original.temperature - rowB.original.temperature;
+};
+
+// Columns initialisation
+const columnHelper = createColumnHelper<WeatherDataResponse>();
+export const getColumns = (
+    title: string,
+    handleFavouriteButton: (stationId: number) => void,
+    isStationFavourite: (stationId: number) => boolean
+) => {
+    return [
+        columnHelper.accessor((row) => row.weather_station_id.id, {
```

Comment:
I would also move this column to the end of the table. In mobile view, the user seems to lose quite valuable real estate for this single column. On top of that, usually actions are located at the end of rows in data tables.